server/src/test/java/org/opensearch/index/translog/transfer/BlobStoreTransferServiceTests.java

```diff
@@ -87,7 +96,8 @@ public void onResponse(FileSnapshot.TransferFileSnapshot fileSnapshot) {
                 public void onFailure(Exception e) {
                     throw new AssertionError("Failed to perform uploadBlobAsync", e);
                 }
-            }, latch)
+            }, latch),
+            WritePriority.HIGH
```

Comment:
Would this be sufficient? Should we rate-limit segment uploads to ensure that network bandwidth doesn't choke up

Reply:
Aren't we doing it in remote store back pressure?

Reply:
On S3 side we haven't configured a hard limit on the bandwidth but we have limited number of connections plus a threadpool which limits stream processing ops to limit amount of cpu consumed which indirectly limits S3 throughput. I believe that we will hit cpu much before we hit network bandwidth.

Reply:
It depends for instance, if there are competing force merges that generates bigger segments, whose upload chokes network bandwidth, indexing request requiring translog uploads can get massively slowed down

Reply:
It can happen if speed of compute is much faster than speed of transfer. By compute I am referring to all the stream buffer processing work currently being done before the transfer of a buffer. This compute happens in a threadpool today which is restricted to 1/8th of total cores for segments.
But I am thinking now that in normal cases speed of compute may be slower but there can be issues on s3 side which can still lead to such issue. In that case we would need an additional rate limiter or segregate s3 client itself.

Reply:
Will prefer a rate limiter on S3 segments uploads

Reply:
https://github.com/opensearch-project/OpenSearch/issues/8619